lib/baseFactory.js

```diff
@@ -55,13 +55,21 @@ class BaseFactory {
             table: this.table,
             params: {
                 id,
-                data: config
+                data: {}
             }
         };
 
+        // Filter for valid keys
+        this.model.allKeys.forEach((key) => {
+            if (config[key]) {
+                modelConfig.params.data[key] = config[key];
+            }
+        });
+
         return nodeify.withContext(this.datastore, 'save', [modelConfig])
-            .then(modelData => this.createClass(hoek.applyToDefaults(modelData, {
-                datastore: this.datastore
+            .then(modelData => this.createClass(hoek.applyToDefaults(config, {
```

Comment:
Oh except I see. The point of this PR is not just to sanitize datastore saves, but really to be able to call `this.createClass` with extra parameters